commit 5b033b04bad7dc57a6e33c680a5c4cbba0a53723
Author:     Yuhong Wen <yh_wen@hotmail.com>
AuthorDate: Mon May 23 15:10:50 2022 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Mon May 23 12:10:50 2022 -0700

    Fixed the snapshot log path info. (#592)
    
    * Fixed the snapshot log path info.
    
    * Updated the logging.
---
 nvflare/private/fed/server/server_engine.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 21ffd29f..3c27fde9 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -651,8 +651,12 @@ class ServerEngine(ServerEngineInternalSpec):
             snapshot.completed = completed
 
             self.server.snapshot_location = self.snapshot_persistor.save(snapshot=snapshot)
-        finally:
-            self.logger.info(f"persist the snapshot to: {self.server.snapshot_location}")
+            if not completed:
+                self.logger.info(f"persist the snapshot to: {self.server.snapshot_location}")
+            else:
+                self.logger.info(f"The snapshot: {self.server.snapshot_location} has been removed.")
+        except BaseException as e:
+            self.logger.error(f"Failed to persist the components. {str(e)}")
 
     def restore_components(self, snapshot: RunSnapshot, fl_ctx: FLContext):
         for component_id, component in self.run_manager.components.items():
